src/query/storages/fuse/src/operations/read/native_data_source_deserializer.rs

```diff
@@ -505,6 +524,177 @@ impl NativeDeserializeDataTransform {
         }
         Ok(())
     }
+
+    // TODO(xudong): add selectivity prediction
+    fn bloom_runtime_filter(
+        &mut self,
+        arrays: &mut Vec<(usize, Box<dyn Array>)>,
+    ) -> Result<Vec<MutableBitmap>> {
+        let mut local_arrays = vec![];
+        // Check if already cached runtime filters
+        if self.cached_bloom_runtime_filter.is_none() {
+            let bloom_filters = self.ctx.get_bloom_runtime_filter_with_id(self.table_index);
+            let bloom_filters = bloom_filters
+                .into_iter()
+                .filter_map(|filter| {
+                    let name = filter.0.as_str();
+                    // Some probe keys are not in the schema, they are derived from expressions.
+                    self.src_schema
+                        .index_of(name)
```

Comment:
There is a case that is not quite discoverable: the probe key is an inner column of a nested type. In another word, `name` could be "a:b:c" or something like that. It's better to add a test case to cover this case.

Reply:
We don't support nested type for probe key, https://github.com/datafuselabs/databend/pull/13993/files#diff-3a9124588945cc9f4c83fd0bfef409c1fc3ef74b2ccfd49e2504af36bbbb3a9dR802